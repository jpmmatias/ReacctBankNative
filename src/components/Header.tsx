import React from 'react';
import {
	StyleSheet,
	Text,
	View,
	TouchableOpacity,
	Dimensions,
  PixelRatio,
} from 'react-native';
import { FontAwesome5 } from '@expo/vector-icons';
const { width, height } = Dimensions.get('screen');
import { IHeader } from '../types';
const Header = ({ name, onPress }: IHeader) => {
	return (
		<View style={styles.container}>
			<Text style={styles.text}>Ol√° {name}</Text>
			<TouchableOpacity onPress={onPress}>
				<FontAwesome5 name='user-circle' size={24} color='white' />
			</TouchableOpacity>
		</View>
	);
};

export default Header;

const styles = StyleSheet.create({
	container: {
		width: PixelRatio.getPixelSizeForLayoutSize(width * 0.35),
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		marginVertical: PixelRatio.getPixelSizeForLayoutSize((height * 2) / 100),
		paddingHorizontal: (width * 7) / 100,
	},
	text: {
		fontFamily: 'Roboto-Medium',
		fontSize: 22,
		color: '#fff',
	},
});
