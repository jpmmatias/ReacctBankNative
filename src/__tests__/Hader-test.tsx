import React from 'react';
import { View } from 'react-native';
import { create } from 'react-test-renderer';
import Header from '../components/Header';

const tree = create(
	<Header
		name='User'
		openDrawer={() => {
			console.log('opend drwer');
		}}
		goToHome={() => {
			console.log('go to home');
		}}
	/>
);

test('Home', () => {
	expect(tree).toMatchSnapshot();
});
