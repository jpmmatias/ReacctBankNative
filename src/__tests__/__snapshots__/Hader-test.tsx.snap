// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 66.7,
        "marginTop": 66.7,
        "paddingLeft": 52.5,
        "paddingRight": 52.5,
        "width": 750,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Roboto-Medium",
          "fontSize": 22,
        },
      ]
    }
  >
    Ol√° 
    User
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
